config/rbac/aggregate_to_edit.yaml

```diff
@@ -0,0 +1,15 @@
+# Automatically aggregated to clusterRoles: edit -> admin
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  labels:
+    rbac.authorization.k8s.io/aggregate-to-edit: "true"
+  name: edit
+rules:
+- apiGroups:
+  - hnc.x-k8s.io
+  resources:
```

Comment:
current admin-role has `*` so `hncconfigurations` too

Reply:
why hncconfigurations too?
isnt its clusterWide object?
you mean `hierarchyconfigurations`?

Reply:
indded :)

Reply:
I dont have practice with `hierarchyconfigurations` so I dont know the consequences, a bit afraid to add something I dont know

Reply:
Ok from what I see this CR allows to add annotations/labels to current namespace and his children.  
This roleBinding is aggregated to edit/admin permission, user that has edit/admin shouldn't be able to modify the namespace including annotations/labels.  

If `hierarchyconfigurations` would influence only the children namespaces then yes I would add it, but because it effects the parent namespace as well, I consider it as a security breach because edit/admin on the namespace shouldnt have permissions to modify the parent namespace.

Sorry I wont add it to the list, if there will be a way to configure `hierarchyconfigurations` only to effect the children (maybe from `hncConfiguration` or something) then yes we can add it as default edit permission.

But thank you for the feedback!

Reply:
then you need to add back the admin-role which [included it and aggregated it to admin](it https://github.com/kubernetes-sigs/hierarchical-namespaces/pull/318/files#diff-bc466b58edebab97ebe341a6e371ee1c6ac9a93ce2b783c90c3e267c05284f8cL6-L18)

Reply:
I see, so it was already configured like this before, thank you for mentioning

@adrianludwin why user with admin role has permissions to edit the namespace labels/annonations?
By default admin cant change anything about the namespace object.

Reply:
It might make sense to distinguish between `hnc-edit`, which gives permission to the namespaced objects (`SubnamespaceAnchor`, `HierarchyConfiguration`, `HierarchicalResourceQuota`) and `hnc-admin`, which would add permission to administer HNC itself (`HNCConfiguration`). That is, `hnc-edit` would let users modify any part of the namespace hierarchy, while `hnc-admin` would let users change the way HNC works.

Being able to edit `HierarchyConfiguration` _looks_ like it lets you change the parent of a namespace, but in reality [HNC itself imposes limitations on this](https://github.com/kubernetes-sigs/hierarchical-namespaces/blob/master/docs/user-guide/concepts.md#namespaces-administrators):

> However, being the admin of a namespace does not give you free reign over that namespace’s config. For example, if you are the admin of a child namespace but not its parent, you may not change the parent of your namespace, as this would remove privileges from the admin of the parent (otherwise known as a superadmin, which in HNC's context means an admin of any ancestor namespace). Similarly, even if your namespace is a root, you may not set its parent to any namespace of which you are not currently an admin, since the namespace could then inherit sensitive information such as Secrets.
> 
> In general, to change the parent of your namespace N from A to B, you must have the following privileges:
> 
> * You must be the admin of the highest namespace that will no longer be an ancestor of namespace N after this change, in order to confirm that you are happy to lose your privileges in namespace N.
> * You must be the admin of namespace B, in order to acknowledge that sensitive objects from B may be copied into namespace N.

This is implemented [here](https://github.com/kubernetes-sigs/hierarchical-namespaces/blob/74fe01460083fd410a8fbd728eb255c4d728dbcb/internal/hierarchyconfig/validator.go#L304).

Reply:
admin is supposed to be used with a RoleBinding [ref](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#user-facing-roles), so a user would not be able to use HNCConfiguration which is ClusterScoped anyway:
* view: view resources in namespace
* edit: view+edit resources (except rbac Role/Rolebinding) in namespace
* admin: view+edit resource (incl. Role/RoleBinding) in namespace
* cluster-admin: view+edit anything (resources `*` verbs: `*`) [ref](https://github.com/kubernetes/kubernetes/blob/82bca6304b5365f7df2627ad2a6fb3d4539bf36f/plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go#L199C19-L200) , with clusterrolebinding

Reply:
Sounds great, but my point was that without HNC, when giving someone admin on namespace A, he cannot change the labels nor annotations of that namespace.
But with the help of HNC, the admin user can configure hiearchyConfiguration to add annotations including on self namespace A.

So its kind of a workaround to be able to manage the namesoace labels+annotations without actually having the permissions.

But now as I think, why not to actually give the ability for the user to label or annotate his namespace even if he cant modify the object? (as long as he is admin)
I bet its not allowed as default only because labels+annotations are part of the object, and its impossible to give permissions only on a specific fields in the object.

Reply:
> But yes, granting someone hnc-admin at the cluster level would allow them to modify any managed label/annotation on any namespace; that's by design.

Thats the point, if someone is granted at not cluster level (only namespaced) he can still modify the annotations of his namespace.

Simple user has only roleBinding and can create hierarchyConfiguration, HNC will create the annotation for him even if he doesn't has permissions.

Anyways I think if someone has admin on the namespace it makes sense that he could add labels to the namespace.

Give me 30 min I will update the PR